code/datums/ammo/xeno.dm

```diff
@@ -0,0 +1,513 @@
+		return //endurance 5 makes you immune to weak neurotoxin
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.chem_effect_flags & CHEM_EFFECT_RESIST_NEURO || H.species.flags & NO_NEURO)
+			H.visible_message(SPAN_DANGER("[M] shrugs off the neurotoxin!"))
+			return //species like zombies or synths are immune to neurotoxin
+
+	if(!isxeno(M))
+		if(insta_neuro)
+			if(M.knocked_down < 3)
+				M.adjust_effect(1 * power, WEAKEN)
+			return
+
+		if(ishuman(M))
+			M.apply_effect(2.5, SUPERSLOW)
+			M.visible_message(SPAN_DANGER("[M]'s movements are slowed."))
+
+		var/no_clothes_neuro = FALSE
+
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(!H.wear_suit || H.wear_suit.slowdown == 0)
+				no_clothes_neuro = TRUE
+
+		if(no_clothes_neuro)
+			if(M.knocked_down < 5)
+				M.adjust_effect(1 * power, WEAKEN) // KD them a bit more
+				M.visible_message(SPAN_DANGER("[M] falls prone."))
+
+/proc/apply_scatter_neuro(mob/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(skillcheck(M, SKILL_ENDURANCE, SKILL_ENDURANCE_MAX))
+			M.visible_message(SPAN_DANGER("[M] withstands the neurotoxin!"))
+			return //endurance 5 makes you immune to weak neuro
+		if(H.chem_effect_flags & CHEM_EFFECT_RESIST_NEURO || H.species.flags & NO_NEURO)
+			H.visible_message(SPAN_DANGER("[M] shrugs off the neurotoxin!"))
+			return
+
+		if(M.knocked_down < 0.7) // apply knockdown only if current knockdown is less than 0.7 second
+			M.apply_effect(0.7, WEAKEN)
+			M.visible_message(SPAN_DANGER("[M] falls prone."))
+
+/datum/ammo/xeno/toxin/on_hit_mob(mob/M,obj/projectile/P)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.status_flags & XENO_HOST)
+			neuro_callback.Invoke(H, effect_power, TRUE)
+			return
+
+	neuro_callback.Invoke(M, effect_power, FALSE)
+
+/datum/ammo/xeno/toxin/medium //Spitter
+	name = "neurotoxic spatter"
+	spit_cost = 50
+	effect_power = 1
+
+	shell_speed = AMMO_SPEED_TIER_3
+
+/datum/ammo/xeno/toxin/queen
+	name = "neurotoxic spit"
+	spit_cost = 50
+	effect_power = 2
+
+	accuracy = HIT_ACCURACY_TIER_5*2
+	max_range = 6 - 1
+
+/datum/ammo/xeno/toxin/queen/on_hit_mob(mob/M,obj/projectile/P)
+	neuro_callback.Invoke(M, effect_power, TRUE)
+
+/datum/ammo/xeno/toxin/shotgun
+	name = "neurotoxic droplet"
+	flags_ammo_behavior = AMMO_XENO|AMMO_IGNORE_RESIST
+	bonus_projectiles_type = /datum/ammo/xeno/toxin/shotgun/additional
+
+	accuracy_var_low = PROJECTILE_VARIANCE_TIER_6
+	accuracy_var_high = PROJECTILE_VARIANCE_TIER_6
+	accurate_range = 5
+	max_range = 5
+	scatter = SCATTER_AMOUNT_NEURO
+	bonus_projectiles_amount = EXTRA_PROJECTILES_TIER_4
+
+/datum/ammo/xeno/toxin/shotgun/New()
+	..()
+
+	neuro_callback = CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(apply_scatter_neuro))
+
+/datum/ammo/xeno/toxin/shotgun/additional
+	name = "additional neurotoxic droplets"
+
+	bonus_projectiles_amount = 0
+
+/*proc/neuro_flak(turf/T, obj/projectile/P, datum/callback/CB, power, insta_neuro, radius)
```

Comment:
if this is all commented out should it not be removed?

Reply:
removed